src/Stratis.Bitcoin.Features.BlockStore/BlockStoreLoop.cs

```diff
@@ -170,31 +170,19 @@ public virtual string StoreName
         }
 
         /// <summary>
-        /// Adds a block to Pending Storage
+        /// Adds a block to Pending Storage.
         /// <para>
-        /// The <see cref="BlockStoreSignaled"/> calls this method when a new block is available. Only add the block to pending storage if:
+        /// The <see cref="BlockStoreSignaled"/> calls this method when a new block is available. Only add the block to pending storage if the store's tip is behind the given block.
         /// </para>
-        /// <list>
-        ///     <item>1: The block does exist on the chain.</item>
-        ///     <item>2: The store's tip is behind the given block.</item>
-        /// </list>
         /// </summary>
-        /// <param name="block">The block to add to pending storage</param>
+        /// <param name="blockPair">The block & chained block pair to be added to pending storage.</param>
         /// <remarks>TODO: Possibly check the size of pending in memory</remarks>
-        public void AddToPending(Block block)
+        public void AddToPending(BlockPair blockPair)
         {
-            uint256 blockHash = block.GetHash();
-            this.logger.LogTrace("({0}:'{1}')", nameof(block), blockHash);
-
-            ChainedBlock chainedBlock = this.Chain.GetBlock(blockHash);
-            if (chainedBlock == null)
-            {
-                this.logger.LogTrace("(-)[REORG]");
-                return;
-            }
+            this.logger.LogTrace("({0}:'{1}')", nameof(blockPair.Block), blockPair.ChainedBlock.HashBlock);
```

Comment:
`this.logger.LogTrace("({0}:'{1}')", nameof(blockPair), blockPair.ChainedBlock);`